### PR TITLE
New improvements

### PR DESCRIPTION
**Pull request by Shqrdx**
Changes
-SetDataOptions,SpriteSortMode,StencilOperation comments / licenses fixes
-New test EnumConformingTest which will intended to test enums on XNA compatibility. 
 I suggest to gradually remove unnecessary enum defines like "Discard = 1," and use this test instead. 
-GameName default name in Universal template changed to Game
-Removed TextureUsage unused enum
-SpriteEffect(not SpriteEffect(s)) is not used at all, its not exist in XNA and I made it internal.

### CODE REVIEW

File: MonoGame.Framework/Graphics/Effect/SpriteEffect.cs [hunk 1]
```diff
@@ -1,24 +1,13 @@
-#region File Description
-//-----------------------------------------------------------------------------
-// SpriteEffect.cs
-//
-// Microsoft XNA Community Game Platform
-// Copyright (C) Microsoft Corporation. All rights reserved.
-//-----------------------------------------------------------------------------
```
Comment (tomspilman) [inline MonoGame.Framework/Graphics/Effect/SpriteEffect.cs]: @KonajuGames - Not sure how we should deal with this copyright?  Don't think we can/should remove it.

Comment (Shqrdx) [inline MonoGame.Framework/Graphics/Effect/SpriteEffect.cs]: What write instead of it ?

Comment (tomspilman) [inline MonoGame.Framework/Graphics/Effect/SpriteEffect.cs]: Not sure yet.

This and the other stock effects came from one of the Microsoft samples.  So we want to leave MS's copyright there.

However we can probably clean it up... remove the `------` crap.  Remove the redundant `// SpriteEffect.cs`.

Not sure how much else can change.

Comment (Shqrdx) [inline MonoGame.Framework/Graphics/Effect/SpriteEffect.cs]: Done !

File: MonoGame.Framework/Graphics/Effect/SpriteEffect.cs [hunk 2]
```diff
@@ -1,24 +1,12 @@
-#region File Description
-//-----------------------------------------------------------------------------
-// SpriteEffect.cs
-//
 // Microsoft XNA Community Game Platform
 // Copyright (C) Microsoft Corporation. All rights reserved.
-//-----------------------------------------------------------------------------
```
Comment (tomspilman) [inline MonoGame.Framework/Graphics/Effect/SpriteEffect.cs]: @KonajuGames  - Can/should we add the MonoGame copyright in here too?  We have made lots of modifications past the stock sample.  Does that change the copyright?

**Comment by Shqrdx** [comment]
Whats up with bot ? Why it is not checks ?

**Comment by tomspilman** [comment]
The mac build failed...

```
error CS2001: Source file `/Applications/buildAgent/work/f26e5a04a66fcbdb/MonoGame.Framework/Graphics/TextureUsage.cs' could not be found
```

File: MonoGame.Framework/Graphics/Effect/SpriteEffect.cs (hunk 2 above)
Comment (Shqrdx) [inline MonoGame.Framework/Graphics/Effect/SpriteEffect.cs]: // Microsoft XNA Community Game Platform
// Copyright (C) Microsoft Corporation & MonoGameTeam. All rights reserved.

)

**Comment by Shqrdx** [comment]
@tomspilman Whats with other stuff, is it ok for merging ?

**Comment by tomspilman** [comment]
I have a pretty busy day today... so this probably won't get merged till tomorrow.

File: MonoGame.Framework/Graphics/Effect/SpriteEffect.cs (hunk 2 above)
Comment (KonajuGames) [inline MonoGame.Framework/Graphics/Effect/SpriteEffect.cs]: If we add our license header, it should be in addition to the unmodified original.

```
// MonoGame - Copyright (C) The MonoGame Team
// This file is subject to the terms and conditions defined in
// file 'LICENSE.txt', which is part of this source code package.
//
// based on
//-----------------------------------------------------------------------------
// SpriteEffect.cs
//
// Microsoft XNA Community Game Platform
// Copyright (C) Microsoft Corporation. All rights reserved.
//-----------------------------------------------------------------------------
```

**Comment by Shqrdx** [comment]
As @KonajuGames said I save both licenses and await your decision.

**Comment by tomspilman** [comment]
Seems ok to me...  i'll merge this today.

**Pull request merged**